### PR TITLE
#256 Replace maven enforcer with antrun copy task

### PR DESCRIPTION
**Pull request by lokm01**
Let's replace the enforcer plugin, which fails the build if the
application.conf is missing from the main resources.

Rather let's materialize the template on the first build, which
will provide reasonable defaults without necessary providing
custom configs while preserving existing files.

This means that people should be able to clone and build the
project (incl tests) out of the box.

Signed-off-by: Jan Scherbaum <kmoj02@gmail.com>

### CODE REVIEW

File: conformance/pom.xml [hunk 1]
```diff
@@ -131,30 +131,21 @@
...
-Please, copy the template file 'application.conf.template' into 'application.conf' and make changes
-corresponding to your use case.
-                                    </message>
-                                </requireFilesExist>
-                            </rules>
-                            <fail>true</fail>
+                            <target>
+                                <copy file="${project.basedir}/src/main/resources/application.conf.template" tofile="${project.basedir}/src/main/resources/application.conf" failonerror="false" overwrite="false" />
+                            </target>
+                            <failOnError>false</failOnError>
```
Comment (benedeki) [inline conformance/pom.xml]: Don't we want to fail in case the copy fails? (of course unless the file was there before)

**Comment by benedeki** [comment]
While I had raised this question originally as well and I am the proponent of "out of the box" executable, i wonder if some more sophisticated solution wouldn't be cool in future (also based on the discussions we had).
1) Template defaults might not be generally suitable
2) Later versions might add some new configurations, and a so far working app suddenly stops functioning correctly.

File: conformance/pom.xml [hunk 2]
```diff
@@ -131,30 +131,21 @@
...
-                                    <message>/src/main/resources/application.conf not found.
-To build Conformance module it is required to have 'application.conf' in the resources directory.
-Please, copy the template file 'application.conf.template' into 'application.conf' and make changes
-corresponding to your use case.
-                                    </message>
-                                </requireFilesExist>
-                            </rules>
-                            <fail>true</fail>
+                            <target>
+                                <copy file="${project.basedir}/src/main/resources/application.conf.template" tofile="${project.basedir}/src/main/resources/application.conf" failonerror="false" overwrite="false" />
```
Comment (Zejnilovic) [inline conformance/pom.xml]: what about spline.properties?

File: menas/pom.xml [hunk 3]
```diff
@@ -260,30 +260,21 @@
...
-                                    <message>/src/main/resources/application.properties not found.
-To build Menas it is required to have 'application.properties' in the resources directory.
-Please, copy the template file 'application.properties.template' into 'application.properties' and make changes
-corresponding to your use case.
-                                    </message>
-                                </requireFilesExist>
-                            </rules>
-                            <fail>true</fail>
+                            <target>
+                                <copy file="${project.basedir}/src/main/resources/application.conf.template" tofile="${project.basedir}/src/main/resources/application.conf" failonerror="false" overwrite="false" />
```
Comment (Zejnilovic) [inline menas/pom.xml]: Menas template is not in a runnable state out of the box.

File: standardization/pom.xml [hunk 4]
```diff
@@ -147,30 +147,21 @@
...
-                                    <message>/src/main/resources/application.conf not found.
-To build Standardization module it is required to have 'application.conf' in the resources directory.
-Please, copy the template file 'application.conf.template' into 'application.conf' and make changes
-corresponding to your use case.
-                                    </message>
-                                </requireFilesExist>
-                            </rules>
-                            <fail>true</fail>
+                            <target>
+                                <copy file="${project.basedir}/src/main/resources/application.conf.template" tofile="${project.basedir}/src/main/resources/application.conf" failonerror="false" overwrite="false" />
```
Comment (Zejnilovic) [inline standardization/pom.xml]: Again, what about the spline.properties?

File: conformance/pom.xml (hunk 1 above)
Comment (lokm01) [reply conformance/pom.xml]: Agree. Changed.

**Comment by lokm01** [comment]
> 2\. Later versions might add some new configurations, and a so far working app suddenly stops functioning correctly.

Yeah if we can come up with something more sophisticated (for updating existing configs automatically) then we definitely should.

I guess this is less evil than failing the build out of the box for now.

File: menas/pom.xml (hunk 3 above)
Comment (GeorgiChochov) [reply menas/pom.xml]: @lokm01 Let's update the `za.co.absa.enceladus.migrations.dump.filepath` as part of this task, it's the only property that seems to me, not set to a reasonable default. I'd suggest: `/tmp/mongo-backups/menas`.

Let's double check if there are others

File: menas/pom.xml [hunk 5]
```diff
@@ -260,30 +260,20 @@
...
-                                    <message>/src/main/resources/application.properties not found.
-To build Menas it is required to have 'application.properties' in the resources directory.
-Please, copy the template file 'application.properties.template' into 'application.properties' and make changes
-corresponding to your use case.
-                                    </message>
-                                </requireFilesExist>
-                            </rules>
-                            <fail>true</fail>
+                            <target>
+                                <copy file="${project.basedir}/src/main/resources/application.conf.template" tofile="${project.basedir}/src/main/resources/application.conf" overwrite="false" />
```
Comment (GeorgiChochov) [inline menas/pom.xml]:
```suggestion
                                <copy file="${project.basedir}/src/main/resources/application.properties.template" tofile="${project.basedir}/src/main/resources/application.properties" overwrite="false" />
```
Menas uses `.properties`

Comment (lokm01) [reply menas/pom.xml]: Thanks!

**Pull request merged**